### DISCUSSION TITLE
SPM을 통한 패키지 분리 이외에 모듈화를 할 수 있는 좋은 방법이 있을까요?

### DISCUSSION THREAD

**Discussion by bazinga94** [nsoojin/MiniSuperApp-fastcampus#13] [아무 질문이나 환영합니다] [answered]
안녕하세요. 슈퍼앱 강의 잘 들었습니다.
모듈화에 대한 강의를 보고 앱에서 모듈화가 가능한 부분에 대해 SPM을 활용하여 패키지 단위로 분리를 하는 상황에서 궁금한 점이 생겨 질문 남깁니다.

모듈화를 할 때 패키지 단위(SPM)로 분리하는 것 이외에 다른 좋은 방법이 있을까요?
앱을 개발하다 보면 서드파티 솔루션이나 오픈소스 라이브러리를 사용하는 경우가 있는데요.
패키지 단위로 분리하여 의존성을 끊으려고 하는 중에 몇몇 소스가 공개되지 않은 솔루션들이 xcframework를 제공하지 않는것을 알게 되었습니다.
확인한 결과 SPM에서는 .a 파일이나 .framework 같은 타입에 대해 지원 하지 않는 것으로 알고 있는데요
(https://betterprogramming.pub/how-to-add-compiled-frameworks-in-swift-package-manager-f0cdbde84fc7)

이런 경우 폴더를 나누거나 접근제어 잘 사용하는것 이외에 다른 방법으로 계층을 나누거나 의존성을 분리할 수 있는 방법이 있을까요??
아니면 SPM을 활용하여 해결이 가능한 상황인지 궁금합니다.
감사합니다.

**Comment by nsoojin** [comment, marked as answer]
안녕하세요. 
SPM은 단지 모듈들을 관리하는 툴입니다. 게다가 비교적 최신에 등장한 툴이고요.
SPM에서 관리할 수 없는 경우라면 Xcode Project에서 프레임워크/라이브러리 타겟을 추가하는 전통적인 방법으로 모듈을 만들어서 모듈화를 하시면 됩니다. 

<img width="741" alt="Screen Shot 2022-03-17 at 6 37 29 PM" src="https://user-images.githubusercontent.com/3890050/158905619-861d29a6-d3b5-4d77-b5e5-9a4333c49434.png">
